feat(load): Add objects to specify people, lighting and equipment

chriswmackey opened the pull request:
This PR also contains a number of other tweaks and cleanups made along the way including:
* ensuring that the code of the ScheduleFixedInterval obeys the max character limit
* fixing some bugs in the setup of the schedule lib
* fixing a bug in the locking of constructions/schedules assigned to Shades

@mostaphaRoudsari ,
If you are able to review this relatively quickly (before the end of today), we can hopefully merge it in soon and this will make my PRs overall a bit more bite-size.  Otherwise, I may add the rest of the load objects (infiltration, ventilation and ProgramType) to this PR tomorrow.

mostaphaRoudsari commented on honeybee_energy/load/lighting.py:
```diff
@@ -0,0 +1,311 @@
+
+
+@lockable
+class Lighting(_LoadBase):
```
would subclassing from `_EquipmentBase` save you writing some of the code here?

chriswmackey replied:
I thought about this but there's really only two properties that Equipment and Lighting have perfectly in common (`watts_per_area` and `schedule`).  Everything else would have to be different and I think I save myself from the most lines of similar code by making `_EquipmentBase` really good at addressing the needs of both `Equipment` objects rather than trying to get it to work for lighting and both equipment objects.

chriswmackey commented on honeybee_energy/lib/_loadtypelimits.py:
```diff
@@ -0,0 +1,19 @@
+"""Load all schedules and schedule types from the IDF libraries."""
```
Right! Thank you for catching that one.

chriswmackey commented on honeybee_energy/programtype.py:
```diff
@@ -5,34 +5,38 @@
-        self.gas_equipment_load = gas_equipment_load
-        self.infiltration_load = infiltration_load
-        self.ventilation_load = ventilation_load
+        self.people = people
```
We will definitely have checks.  Either the next PR or the one after it will include them here.  I'm just trying to break these major feature PRs down into bite-size chunks for review without breaking the tests so that is why there's still a bit of paceholder code here.

chriswmackey commented on honeybee_energy/lib/schedules.py:
```diff
@@ -0,0 +1,36 @@
+# and auto-generate materials if they were not loaded from default.idf
+try:
+    seated_activity = _idf_schedules['Seated Adult Activity']
+except KeyError:
```
I definitely don't expect this to be the normal situation (the `except` here is truly exceptional). However, this code is here because we imagined allowing users to edit the default ,idf libraries, which now include the default constructions, materials, schedule types, and schedules.  So this code here is to catch the case that a user totally deletes this IDF schedule object that is critical for the rest of the library to run.  This way, the whole plugin doesn't crash if someone edits the IDF libraries incorrectly.

mostaphaRoudsari replied:
Didn't we decide to separate the library files from the user one? similar to what we currently do in the legacy. For the exact same reasons that you mentioned I think the file that user edits should be separate from the file that we load as the library.

chriswmackey replied:
Yes, there are two separate IDF files for `default.idf` and `user_library.idf`, which you can see here:
https://github.com/ladybug-tools/honeybee-energy/tree/master/honeybee_energy/lib/idf/constructions

And users could also add more idf files into that folder if they like and have them imported to the library (like if someone has a glazing construction that they exported from LBNL WINDOW, they can just drop it in that folder). In any case, having two separate files doesn't change the fact that the user can edit either depending on whether they want to change the defaults or just add new stuff on top of the defaults.  So my previous comment still holds and I am not sure how having separate files affects the fact that the we should still have a recovery system if the user edits the defaults incorrectly.

chriswmackey commented on honeybee_energy/lib/schedules.py:
```diff
@@ -0,0 +1,36 @@
+
+
+# methods to look up schedule types from the library
+
```
My PEP8 filters were throwing me an error when there weren't two blank lines before a function.  Do you think that I should just ignore this for cases like this?

mostaphaRoudsari replied:
I would say so.

chriswmackey commented on honeybee_energy/load/_base.py:
```diff
@@ -0,0 +1,78 @@
+    """A base object for all load definitions.
+
+    Properties:
+        * name
```
I realized that this is what I should do in order to get bullet lists to display correctly in the sphinx docs.   Otherwise, we get something that looks like [what you see under properties here](https://www.ladybug.tools/honeybee-energy/docs/honeybee_energy.construction.html). Of course, this is a hidden module so it doesn't really matter too much here but I'm trying to follow this as a good practice convention across all my docstrings.

mostaphaRoudsari replied:
I see. Nice. Another item to add to the issue that you have for documentation.

chriswmackey commented on honeybee_energy/load/equipment.py:
```diff
@@ -0,0 +1,509 @@
+        return equipment, zone_name
+
+    @classmethod
+    def from_dict(cls, data):
```
That's a fair point. I used a helper for the `to_dict`, `from_idf` and `to_idf` methods so I might as well do the same and save a few lines of code here.

chriswmackey commented:
Thank you for the quick review @mostaphaRoudsari .  I addressed most of your comments with edits to the code and I left the ones where I just responded to your comment unresolved so that you can see my reply.  I'm going to merge this in and continue working towards the next PR.
